hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/utils/ActionStack.java

```diff
@@ -214,7 +214,12 @@ private ContractAction finalFormOf(@NonNull final ContractAction action, @NonNul
                     final var haltReason = maybeHaltReason.get();
                     builder.error(Bytes.wrap(haltReason.name().getBytes(UTF_8)));
                     if (CALL.equals(action.callType()) && haltReason == INVALID_SOLIDITY_ADDRESS) {
-                        allActions.add(new ActionWrapper(helper.createSynthActionForMissingAddressIn(frame)));
+                        final var invalidAddressContext = FrameUtils.invalidAddressContext(frame);
+                        // Only create the synth action if the invalid address was actually a call target
+                        if (invalidAddressContext != null && invalidAddressContext.raisedDueToInvalidCallTarget()) {
```

Comment:
There is still a bit of fragility here that you described for the previous solution because all of this depends on code in the custom operations to set the correct state for this context variable.  I would prefer some sort of log message if `invalidAddressContext == null` because we never really expect it to be null in this if block.

Reply:
sure but what happens if it should be null?  We get a NPE?  That might be good because then we would at least be alerted that something out of the ordinary has happened.